tests/unit_tests/test_helm_chart.py

```diff
@@ -305,3 +293,71 @@ def group_manifests(ungrouped: Iterable[Mapping[str, Any]]) -> GroupedManifests:
             )
         group[name] = manifest
     return groups
+
+
+def test_init_container_config_not_available_when_disabled():
+    manifests = render_chart(
+        values={
+            "resources": HIGH_RESOURCES,
+            "initResources": LOW_RESOURCES,
+            "initContainer": {"enabled": False},
+        }
+    )
+
+    assert (
+        (
+            manifests["StatefulSet"]["blueapi"]["spec"]["template"]["spec"][
+                "initContainers"
+            ]
+        )
+        is None
+    )
+
+    config = yaml.safe_load(
+        manifests["ConfigMap"]["blueapi-config"]["data"]["config.yaml"]
+    )
+
+    assert config.get("scratch", None) is None
+
+    manifests = render_chart(
+        values={
+            "resources": HIGH_RESOURCES,
+            "initContainer": {"enabled": False},
+        }
+    )
+    assert (
+        manifests["StatefulSet"]["blueapi"]["spec"]["template"]["spec"][
+            "initContainers"
+        ]
+        is None
+    )
+
+    config = yaml.safe_load(
+        manifests["ConfigMap"]["blueapi-config"]["data"]["config.yaml"]
+    )
+
+    assert config.get("scratch", None) is None
+
+
+def test_init_container_config_copied_to_worker_when_enabled():
```

Comment:
Could: Also have a test that proves the scratch config is appended, not written over, the worker config